lib/VCMIDirs.cpp

```diff
@@ -164,178 +97,111 @@ class VCMIDirsWIN32 final : public IVCMIDirs
 		std::string libraryName(const std::string& basename) const override;
 
 		void init() override;
+
 	protected:
-		bfs::path oldUserDataPath() const;
-		bfs::path oldUserSavePath() const;
+		mutable std::optional<JsonNode> dirsConfig;
```

Comment:
Why it needs to be mutable? Why you need optional?

Might be better to drop mutable / optional, and always attempt to load config on construction. And use default-init empty JsonNode if config file does not exists